modules/log/log.go

```diff
@@ -174,7 +177,7 @@ type logMsg struct {
 // it can contain several providers and log message into all providers.
 type Logger struct {
 	adapter string
-	lock    sync.Mutex
+	lock    sync.RWMutex
```

Comment:
I think It's for the outputs.

Normally, locks should be per data instead of a chunk of code. And I don't see any other fields except for 'outputs', would be updated concurrently. And channel variables should be thread-safe in its own right.